app/src/main/java/org/stacks/app/domain/Login.kt

```diff
@@ -2,32 +2,44 @@ package org.stacks.app.domain
 
 import com.colendi.ecies.EncryptedResult
 import com.colendi.ecies.Encryption
+import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.flow.Flow
 import kotlinx.coroutines.flow.flatMapConcat
 import kotlinx.coroutines.flow.flow
 import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.withContext
+import org.blockstack.android.sdk.model.BlockstackIdentity
 import org.blockstack.android.sdk.toBtcAddress
-import org.stacks.app.data.Auth
+import org.json.JSONException
+import org.json.JSONObject
+import org.kethereum.bip32.generateChildKey
+import org.kethereum.bip32.model.ExtendedKey
+import org.kethereum.bip32.toKey
+import org.kethereum.bip39.model.MnemonicWords
+import org.kethereum.bip39.toSeed
+import org.kethereum.extensions.toHexStringNoPrefix
+import org.komputing.kbip44.BIP44Element
 import org.stacks.app.data.EncryptedPreferencesIdentityRepository
 import org.stacks.app.data.EncryptedPreferencesSecretKeyRepository
 import org.stacks.app.data.IdentityModel
-import org.stacks.app.data.network.GaiaHubService
+import org.stacks.app.data.network.services.GaiaService
+import org.stacks.app.shared.IdentitiesParsingFailed
 import org.stacks.app.shared.mapIfSuccess
 import org.stacks.app.shared.toResult
+import timber.log.Timber
 import javax.inject.Inject
 
 class Login
 @Inject constructor(
-    private val gaiaHubService: GaiaHubService,
-    private val auth: Auth,
+    private val gaiaService: GaiaService,
     private val identitiesRepo: EncryptedPreferencesIdentityRepository,
     private val secretKeyRepo: EncryptedPreferencesSecretKeyRepository
 ) {
 
     fun login(mnemonicWords: String): Flow<Result<List<IdentityModel>>> =
```

Comment:
Could be a simple `suspend` fun, does not need to return a flow, it's a simple action.